### DISCUSSION TITLE
App Improvement Proposals

### DISCUSSION THREAD

**Discussion by rramirez-dev** [mikaelacaron/Basic-Car-Maintenance#331] [Ideas]
1. Prompt the user to add a vehicle when the app launches because you can't add an event without a vehicle.
2. Improve the Vehicle and Event displays.
![DisplayExamples](https://github.com/user-attachments/assets/3056b56f-1e52-4454-bbc1-8d9c19064191)

**Comment by mikaelacaron** [comment]
1: is happening with #285 
2: I'm cool with this being two separate issues (be sure to split the screenshot in each issue (because it can be confusing to see them both together)

**Reply by rramirez-dev** [reply]
I was just throwing the ideas out there so I put them in one screenshot. If they are approved I can put the relevant screenshots in two separate tickets.

**Comment by rramirez-dev** [comment]
The issues have been created. They are open to be assigned to anyone.

IMPROVE -Vehicle Display in settings:  https://github.com/mikaelacaron/Basic-Car-Maintenance/issues/332

IMPROVE-How events are displayed on the dashboard with and without notes: https://github.com/mikaelacaron/Basic-Car-Maintenance/issues/333

**Comment by maartinj** [comment]
Add clear button to fields, where possible. To easily clear the current field of text instead of holding delete button:

<img src="https://github.com/user-attachments/assets/d9ae5616-9843-4b56-909e-16a143a3561f" height="200">

**Reply by mikaelacaron** [reply]
Great idea! let me make this issue, because it can be a bunch of separate issues
is there a recommended way to do this? (that's a best practice) so that everyone implements it the same way?

**Reply by maartinj** [reply]
Beginners can use ViewModifier 'TextFieldClearButton', which I defined in PR #301 👍

**Reply by mikaelacaron** [reply]
Does this account for right to left languages? Is there a not a native way to do this in SwiftUI?

**Reply by maartinj** [reply]
To be honest, I do not know whether will work for right to left languages. I will check this.

As far as I know, there is no native way to clear all text in textfield by one click in SwiftUI.
